es2020.md

```diff
@@ -0,0 +1,43 @@
+This document specifies the extensions to the core ESTree AST types to support the ES2020 grammar.
+
+# Literal
+
+```js
+extend interface Literal <: Expression {
+    type: "Literal";
+    value: string | boolean | null | number | RegExp | bigint;
+}
+```
+
+- `value` property can be a `BigInt` value to represent `BigInt` literals
+  such as `100n`.
+
+## BigIntLiteral
+
+```js
+interface BigIntLiteral <: Literal {
+  bigint: string;
+}
+```
+
+- `bigint` property is the string representation of the `BigInt` value.
+  It doesn't include the suffix `n`.
+- In environments that don't support `BigInt` values, `value` property will be
```

Comment:
Instead of saying this in prose, I would add a `value: null` entry to the interface description.

Reply:
The type of `value` property is `string | boolean | null | number | RegExp | bigint`, so it's nullable already.

Reply:
The purpose of this sentence is to describe we cannot use `node.value === null` to distinguish if a node is `null` literal or not.

Reply:
This seems like it could get pretty confusing if there are two possible representations of the field. Wouldn't it be much nicer to do
```
interface BigIntLiteral <: Literal { }
```
with the expectation that `.value` is the string representation when `.type === "BigIntLiteral"`?

Reply:
I agree that this is inconvenient. But we did the same thing in the past: es2015 and es2018 RegExp enhancements. So this is following precedents.